recipes/gtsam/all/conanfile.py

```diff
@@ -99,30 +99,43 @@ def _configure_cmake(self):
             self._cmake.definitions["GTSAM_INSTALL_CPPUNITLITE"] = self.options.install_cppunitlite
             self._cmake.definitions["GTSAM_INSTALL_GEOGRAPHICLIB"] = self.options.install_geographiclib
             self._cmake.definitions["GTSAM_USE_SYSTEM_EIGEN"] = False
-            self._cmake.configure(source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+            self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-            if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
-                raise ConanInvalidConfiguration ("GTSAM requirews MSVC >= 15")
+        if self.settings.compiler == "Visual Studio" and tools.Version(self.settings.compiler.version) < 15:
+            raise ConanInvalidConfiguration("GTSAM requirews MSVC >= 15")
```

Comment:
Why should I check for VS outside Windows?

Reply:
It does not matter because VS is only available for Windows.
Again, it's a matter of opinion. I prefer less inlining. It's up to you.
The outcome will be the same.